static/js/stream_topic_history.js

```diff
@@ -52,7 +52,7 @@ exports.per_stream_history = function (stream_id) {
     */
 
     const topics = new FoldDict();
-
+    const max_message_ids = new Map(); // stream_id -> max_message_ids
```

Comment:
Can you extract the logic to track `max_message_id` and its tests into its own commit?

Reply:
@timabbott done. :)